ffmpeg/ffmpeg.go

```diff
@@ -28,6 +28,11 @@ const (
 	Amd
 )
 
+type ComponentOpts struct {
```

Comment:
I might suggest `ComponentOptions` but I have no issue with the current name.

Reply:
Yes, that would be more consistent with our existing `TranscodeOptions` and `TranscodeOptionsIn` struct naming. Kinda long / wordy, but consistent.

Would really love to find a better word here than "component" though.

Reply:
In M$ Windows BDA-land, we call them "filters" ...  The word "filter" might be a bit confusing to the open source world, however.  What do you think?   When you assemble the correct filters, the resulting pipeline is called a "graph".

Meanwhile, GSTREAMER is another framework that can be used for comparison.  In GSTREAMER, they're called "elements"

I see nothing wrong with the word "components" but maybe "elements" could be used if you're not happy with "components" ...  I am used to "filters" from my past of windows driver dev, but I don't think most users are familiar with the term.

Reply:
Good to know what other projects call them! Agreed that using "filter" may be confusing, especially since we already use libavfilter here.

I'm leaning slightly into "components" over "elements" [1], but could go either way if you have a preference there. For now I've renamed things to "ComponentOptions", but that can be changed.

[1] The term "element" feels a bit more generic than "component", when it'd be good to have something more specific that describes their role in the system. Now that I think of it, it's kind of strange that there isn't an industry-standard term for these pieces of the transcoding pipeline!

Reply:
I like components better than elements, but I don't feel strongly either way.

I think the industry standard is actually the Microsoft term, "filter" ...  However, M$ has become much less relevant in this space lately, and I don't think non-windows programmers like calling it a "filter".

Let's keep it as-is, "ComponentOptions" unless anybody feels differently